### PR TITLE
Fix Safari rendering issues

### PR DESCRIPTION
**Pull request by snowystinger**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/react-aria-components/docs/GridList.mdx [hunk 1]
```diff
@@ -168,6 +169,16 @@ import {GridList, Item, Button} from 'react-aria-components';
       border-start-start-radius: 0;
       border-start-end-radius: 0;
     }
+
+    & .react-aria-Item[aria-selected=true]:has(+ .react-aria-DropIndicator + [aria-selected=true]) {
```
Comment (snowystinger) [inline packages/react-aria-components/docs/GridList.mdx:173]: fixes the case where contiguous selections were being broken up by hidden drop indicators

File: packages/react-aria-components/docs/GridList.mdx [hunk 2]
```diff
@@ -114,6 +114,7 @@ import {GridList, Item, Button} from 'react-aria-components';
     color: var(--text-color);
     font-size: 1.072rem;
     position: relative;
+    transform: translateZ(0);
```
Comment (snowystinger) [inline packages/react-aria-components/docs/GridList.mdx:117]: fixs a safari issue where a gap would appear between two selected rows on a regular dpi screen

File: packages/react-aria-components/docs/Table.mdx [hunk 3]
```diff
@@ -1103,6 +1114,7 @@ function Example() {
 
 .react-aria-DropIndicator[data-drop-target] {
   outline: 1px solid var(--highlight-background);
+  transform: translateZ(0);
```
Comment (snowystinger) [inline packages/react-aria-components/docs/Table.mdx:1117]: drop indicator changes fix issue where drop indicator would not stretch entire width of safari tables

File: packages/react-aria-components/docs/GridList.mdx (hunk 1 above)
Comment (devongovett) [reply packages/react-aria-components/docs/GridList.mdx:173]: Would it make sense to combine these rules with the above ones? Also I think the `&` is not required here.

File: packages/react-aria-components/docs/Table.mdx [hunk 4]
```diff
@@ -216,6 +216,16 @@ import {Table, TableHeader, TableBody, Column, Row, Cell} from 'react-aria-compo
...
+
+    .react-aria-Row[aria-selected=true]:has(+ .react-aria-DropIndicator + [aria-selected=true]) .react-aria-Cell {
+      border-end-start-radius: 0;
+      border-end-end-radius: 0;
+    }
+
+    .react-aria-Row[aria-selected=true] + .react-aria-DropIndicator + [aria-selected=true] .react-aria-Cell {
+      border-start-start-radius: 0;
+      border-start-end-radius: 0;
+    }
```
Comment (devongovett) [inline packages/react-aria-components/docs/Table.mdx:228]: same

File: packages/react-aria-components/docs/Table.mdx [hunk 5]
```diff
@@ -1081,6 +1091,7 @@ function Example() {
 .react-aria-Row {
   &[data-dragging] {
     opacity: 0.6;
+    transform: translateZ(0);
```
Comment (devongovett) [inline packages/react-aria-components/docs/Table.mdx:1094]: Why this one?

Comment (snowystinger) [reply packages/react-aria-components/docs/Table.mdx:1096]: same reason as this one https://github.com/adobe/react-spectrum/pull/4333/files#r1157714584

File: packages/react-aria-components/docs/GridList.mdx (hunk 1 above)
Comment (snowystinger) [reply packages/react-aria-components/docs/GridList.mdx:173]: yeah, & was copied from above lines

File: packages/react-aria-components/docs/Table.mdx [hunk 6]
```diff
@@ -196,6 +196,8 @@ import {Table, TableHeader, TableBody, Column, Row, Cell} from 'react-aria-compo
   }
 
   .react-aria-Cell {
+    transform: translateZ(0);
```
Comment (snowystinger) [inline packages/react-aria-components/docs/Table.mdx:199]: fixed issue where cells that had any border radius would separate visually from their neighbor

**Review by LFDanLu — APPROVED** [review]
LGTM, tested on Chrome, Safari, FF on desktop and mobile

File: packages/react-aria-components/docs/GridList.mdx [hunk 7]
```diff
@@ -159,12 +160,14 @@ import {GridList, Item, Button} from 'react-aria-components';
   @supports selector(:has(.foo)) {
     gap: 0;
 
-    & .react-aria-Item[aria-selected=true]:has(+ [aria-selected=true]) {
+    .react-aria-Item[aria-selected=true]:has(+ [aria-selected=true]),
+    .react-aria-Item[aria-selected=true]:has(+ .react-aria-DropIndicator + [aria-selected=true]) {
```
Comment (devongovett) [inline packages/react-aria-components/docs/GridList.mdx:164]: I guess we should probably change the drop indicator color in this case as well so you can see it but can do that later.

**Review by devongovett — APPROVED** [review]

**Pull request merged**